src/insert-text-at-cursor.js

```diff
@@ -86,7 +86,12 @@ export default function (input, text) {
 
           // Finally insert a new node. The browser will automatically
           // split start and end nodes into two if necessary
-          range.insertNode(textNode)
+          if (range.commonAncestorContainer.nodeName === '#text') {
+            range.insertNode(textNode)
+          } else {
+            var value = input.value
+            input.value = value.slice(0, start) + text + value.slice(end)
+          }
```

Comment:
I wonder if we can move this so there is only one place that we do this?

Reply:
Refactored to avoid code repetition